aptos-move/move-examples/defi/aptoswap/core/sources/liquidity_pool.move

```diff
@@ -0,0 +1,442 @@
+            primary_fungible_store::create_primary_store_enabled_fungible_asset(
+                pool_cref,
+                option::none(),
+                lp_token_name,
+                lp_token_symbol,
+                6,
+                string::utf8(b"http://aptoswap.com/favicon.ico"),
+                string::utf8(b"http://aptoswap.com/")
+            );
+            (
+                fungible_asset::generate_mint_ref(pool_cref),
+                fungible_asset::generate_burn_ref(pool_cref),
+            )
+        };
+
+        let x_store = fungible_asset::create_store(&object::create_object_from_object(pool_signer), x);
+        let y_store = fungible_asset::create_store(&object::create_object_from_object(pool_signer), y);
+
+        // create store reusing the same liquidity pool object to hold min_liquidity
+        fungible_asset::create_store(pool_cref, fungible_asset::mint_ref_metadata(&lp_token_mint_ref));
+        let liquidity_pool = LiquidityPool {
+            extend_ref: generate_extend_ref(pool_cref),
+            store_pair: vector[x_store, y_store],
+            lp_token_mint_ref,
+            lp_token_burn_ref,
+            fee: 0
+        };
+        move_to(pool_signer, liquidity_pool);
+        let map = &mut borrow_global_mut<LiquidityPoolManager>(@pool_manager).map;
+        smart_table::add(map, vector[x, y], object::object_from_constructor_ref<LiquidityPool>(pool_cref));
+    }
+
+    /// Mint new liquidity coins.
+    /// * `coin_x` - coin X to add to liquidity reserves.
+    /// * `coin_y` - coin Y to add to liquidity reserves.
+    /// Returns LP coins: `Coin<LP<X, Y, Curve>>`.
+    public fun mint(
+        coin_x: FungibleAsset,
+        coin_y: FungibleAsset
+    ): FungibleAsset acquires LiquidityPoolManager, LiquidityPool {
+        let x = fungible_asset::metadata_from_asset(&coin_x);
+        let y = fungible_asset::metadata_from_asset(&coin_y);
+        if (!is_valid_pair(x, y)) {
+            return mint(coin_y, coin_x)
+        };
+
+        let pool = borrow_lp(x, y);
+        let x_store = *vector::borrow(&pool.store_pair, 0);
+        let y_store = *vector::borrow(&pool.store_pair, 1);
+        let x_reserve_size = fungible_asset::balance(x_store);
+        let y_reserve_size = fungible_asset::balance(y_store);
+
+        let x_value = fungible_asset::amount(&coin_x);
+        let y_value = fungible_asset::amount(&coin_y);
+
+        let lp_token_supply = lp_token_supply(pool);
+        let provided_liq = if (lp_token_supply == 0) {
+            let initial_liq = (math128::sqrt((x_value as u128) * (y_value as u128)) as u64);
+            let min_liquidity = fungible_asset::mint(&pool.lp_token_mint_ref, MINIMAL_LIQUIDITY);
+            primary_fungible_store::deposit(object::address_from_extend_ref(&pool.extend_ref), min_liquidity);
+            initial_liq - MINIMAL_LIQUIDITY
+        } else {
+            let x_liq = math64::mul_div(x_value, (lp_token_supply as u64), x_reserve_size);
```

Comment:
Are you planning on writing the router code as well? Without it, this is a possible foot gun as you would only receive the smaller amount based on constant product